tasks/main.yml

```diff
@@ -255,7 +255,7 @@
     regexp: "^export HTTP_PROXY="
     line: "export HTTP_PROXY=\"{{docker_http_proxy}}\""
     state: present
-  when: docker_http_proxy is defined and (docker_http_proxy != None)
+  when: docker_http_proxy is defined and (docker_http_proxy != None) and not is_systemd
```

Comment:
I propose to replace this check (and others similarly) by:
```
  when:
  - (docker_http_proxy|default('')) != ''
  - not is_systemd
```
which is a cleaner check for `docker_http_proxy` (that I originally contributed), as I just came across a case where the check didn't work (as expected, on v2.3.1), when defining the variable like this:
```
docker_http_proxy: "{{ proxy_env.http_proxy | default(None) }}"
```